common/src/main/nl/jft/common/util/elo/EloExpectation.java

```diff
@@ -108,4 +108,34 @@ public double getSecondDelta() {
         return secondDelta;
     }
 
+    @Override
+    public int hashCode() {
+        int result = 17;
+        result = 31 * result + Double.hashCode(firstElo);
+        result = 31 * result + Double.hashCode(secondElo);
+        result = 31 * result + firstGoalsScored;
+        result = 31 * result + secondGoalsScored;
+        result = 31 * result + maxGoals;
+        result = 31 * result + Double.hashCode(firstDelta);
+        result = 31 * result + Double.hashCode(secondDelta);
+
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!(obj instanceof EloExpectation)) {
+            return false;
+        } else {
+            EloExpectation other = (EloExpectation) obj;
+
+            return firstElo == other.firstElo &&
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Reduce the number of conditional operators (6) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://92.110.24.119:9000/coding_rules#rule_key=squid%3AS1067)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Equality tests should not be made with floating point values. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://92.110.24.119:9000/coding_rules#rule_key=squid%3AS1244)